### DISCUSSION TITLE
mqtt_room not updating status after receiving mqtt message

### DISCUSSION THREAD

**Discussion by youngwookie** [ESPresense/ESPresense#6] [Q&A] [answered]
It seems like I am able to get MQTT message that contains the fields mqtt_room expects (id / distance, but seems to be missing 'name'), but the mqtt_room doesn't seem to update. I am not sure why it may be.
If it helps, when I used one of the random MAC addresses from the phone with esp32-mqtt-room, the rooms were getting detected, although only for a short while until MAC address changed.
(mqtt_room doesn't have much in the way of debug messages - only one error message ever seems to go to the logger, so I couldn't check much)

Could someone point me to the right direction?

## EDIT
I tried to publish mqtt message manually and found that adding field 'name' that matches the name in configuration.yaml does the trick, though I am not sure how to add the 'name' to espresence.
## EDIT END

I have the following set up:
  - platform: mqtt_room
    state_topic: "espresense/rooms"
    **device_id: "apple:1007:11-8"**
    name: "iphone"

Here's the mqtt message:
Message 1503 received on espresense/rooms/Living_Room at 2:25 AM:
{
    **"id": "apple:1007:11-8",**
    "rssi@1m": -49,
    "rssi": -52,
    "mac": "491c8ef2ad00",
    "raw": 1.22,
    "distance": 1.22,
    "speed": 0
}

**Comment by youngwookie** [comment]
I've no clue why, but copy-pasting the topic from mqtt message into configuration.yaml worked.

**Reply by mjleonides** [reply]
Couple of years later but this just happened to me as well. My bluetooth beacon continuously read not_home despite updates in MQTT Explorer. Same solution worked though, copy and pasting topic from MQTT Explorer into mqtt_room sensor setup in configuration.yaml .

**Comment by DTTerastar** [comment, marked as answer]
I finally got around to making the documentation better for ESPresense.  Check out https://ESPresense.com. It should be more clear now.  I don't include a "name" unless we get one from the announcement (rare), but home-assistant really only needs id to work.

**Reply by youngwookie** [reply]
Yeah after I played around with the mqtt message & BleFingerprint, I learned that maybe there was some ghost character in my configuration.yaml.

All is working well, as long as I can somehow match fingerprint & the mac address to the device (since it is randomized and I can't figure out a way to see "currently advertized address" in iPhone)

Thank you for the updated document - I was trying to cross reference the esp32-mqtt-room project documents before

**Reply by DTTerastar** [reply]
https://espresense.com/home_assistant talks about listening to the serial out and it shows a message: Close or Left so you can take whatever you want to track and put it up to the esp32 and it should show Close, then take it away and it'll say Left.  I do want to add that to mqtt as well so you don't need the serial output.

**Reply by youngwookie** [reply]
The serial output is really helpful to identify the devices, thank you again!
I've now got the two devices at home tracked across 4 rooms :)
